proto-lens-protobuf-types/src/Data/ProtoLens/Any.hs

```diff
@@ -0,0 +1,68 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Data.ProtoLens.Any
+    ( Any
+    , pack
+    , packWithPrefix
+    , unpack
+    , UnpackError(..)
+    ) where
+
+import Control.Exception (Exception(..))
+import Data.Monoid ((<>))
+import Data.Text (Text, breakOnEnd)
+import Data.Typeable (Typeable)
+import Data.ProtoLens
+    ( decodeMessage
+    , def
+    , encodeMessage
+    , Message(..)
+    , MessageDescriptor(..)
+    )
+import Lens.Family2 ((&), (.~), (^.))
+import Proto.Google.Protobuf.Any
+
+-- | Packs the given message into an 'Any' using the default type URL prefix
+-- "type.googleapis.com".
+pack :: forall a . Message a => a -> Any
+pack = packWithPrefix googleApisPrefix
+
+googleApisPrefix :: Text
+googleApisPrefix = "type.googleapis.com"
+
+
+-- | Packs the given message into an 'Any' using the given type URL prefix.
+packWithPrefix :: forall a . Message a => Text -> a -> Any
+packWithPrefix prefix x =
+    def & typeUrl .~ (prefix <> "/" <> name)
+        & value .~ encodeMessage x
+  where
+    name = messageName (descriptor :: MessageDescriptor a)
+
+
+data UnpackError
+    = DifferentType
+        { expectedMessageType :: Text -- ^ The expected @packagename.messagename@
+        , actualUrl :: Text -- ^ The typeUrl in the 'Any' being unpacked
+        }
+    | DecodingError String  -- ^ The error from decodeMessage
```

Comment:
Switched to Text.

I originally used String since that's what we get from `decodeMessage :: Message a => Either String a`.  But since we have a new type, there isn't a strong need for them to be consistent.